eclair-core/src/main/scala/fr/acinq/eclair/payment/PaymentInitiator.scala

```diff
@@ -62,6 +78,18 @@ object PaymentInitiator {
                                 externalId: Option[String] = None,
                                 predefinedRoute: Seq[PublicKey] = Nil,
                                 assistedRoutes: Seq[Seq[ExtraHop]] = Nil,
-                                routeParams: Option[RouteParams] = None)
+                                routeParams: Option[RouteParams] = None) {
+    // We add one block in order to not have our htlcs fail when a new block has just been found.
+    def finalExpiry(currentBlockHeight: Long) = finalExpiryDelta.toCltvExpiry(currentBlockHeight + 1)
+  }
+
+  case class SendPaymentConfig(id: UUID,
+                               parentId: UUID,
+                               externalId: Option[String],
+                               paymentHash: ByteVector32,
+                               targetNodeId: PublicKey,
+                               paymentRequest: Option[PaymentRequest],
+                               storeInDb: Boolean,
```

Comment:
Ok, maybe because we don't want to store child payments, but if so why not check the `parent_id`?

Reply:
For trampoline you never want to store in the DB when you're a relaying node. Or maybe in the future after a refactoring we won't want to store multi-part child payments (we currently do store them when we're the sender).
We could use `parentId != id` as an implicit signal, but I prefer an explicit signal, otherwise users of this class will have to know that this is implicitly inferred and take it into account for their use-case.

Reply:
Ok, let's add a comment then:

```suggestion
                               storeInDb: Boolean, // e.g. for trampoline we don't want to store in the DB when we're relaying payments
```